### PR TITLE
Redesign Website

### PR DESCRIPTION
**Pull request by Willyboar**
Redesign the homepage for start.

### CODE REVIEW

File: templates/base.tmpl [hunk 1]
```diff
@@ -1,50 +1,51 @@
-#? stdtmpl | standard
+##? stdtmpl | standard
```
Comment (FedericoCeratto) [inline templates/base.tmpl]: Why the double ##?

Comment (FedericoCeratto) [reply templates/base.tmpl]: It leads to `templates/base.tmpl(7, 16) Error: invalid indentation`

File: templates/home.tmpl [hunk 2]
```diff
@@ -1,158 +1,72 @@
...
-
-<div class="columns">
-  <div class="column col-12">
-    <p><b>Developer?</b> Search for packages or jump directly to a package page.
+<header class="hero">
+    <img class="logo" src="img/nimble-logo.png" alt="logo">
+    <h1>Nimble</h1>
+    <h5 style="text-align: center;">Nimble Package Directory</h5>
+    ${generate_search_box()}
+    <p>You don't have Nim in your computer. Install it here.</p>
```
Comment (FedericoCeratto) [inline templates/home.tmpl]: This is false.

File: templates/home.tmpl [hunk 3]
```diff
@@ -1,158 +1,72 @@
...
 #proc generate_home_page(most_queried_packages, new_packages: seq[Pkg], github_trending: seq[JsonNode]): string =
 #  result = ""
-
-${generate_search_box()}
-
-<div class="columns">
-  <div class="column col-12">
-    <p><b>Developer?</b> Search for packages or jump directly to a package page.
+<header class="hero">
+    <img class="logo" src="img/nimble-logo.png" alt="logo">
```
Comment (FedericoCeratto) [inline templates/home.tmpl]: This is confusing: it looks like Nimble is the name of the package directory itself.

File: templates/home.tmpl [hunk 4]
```diff
@@ -1,158 +1,72 @@
...
 #
 #proc generate_home_page(most_queried_packages, new_packages: seq[Pkg], github_trending: seq[JsonNode]): string =
 #  result = ""
-
-${generate_search_box()}
-
-<div class="columns">
-  <div class="column col-12">
-    <p><b>Developer?</b> Search for packages or jump directly to a package page.
+<header class="hero">
```
Comment (FedericoCeratto) [inline templates/home.tmpl]: There is a lot of empty space and large titles in the page. I had to zoom out to 60% to make it usable.

File: templates/home.tmpl [hunk 5]
```diff
@@ -1,158 +1,72 @@
...
+      <h2><i class="fas fa-poll" style="color:#2ff36b"></i> Trending on GitHub</h2>
+      # for i in 0 .. min(8, len(github_trending))-1:
+      # let pkg = github_trending[i]
+      <div class="four columns" id="pack-show">
+        <h4>${pkg["name"].str}</h4>
+        <span class="stars"><i class="fas fa-star" style="color:#f7e942"></i> ${pkg["stargazers_count"].getNum}</span>
+        <p class="show-text"><img src="${pkg["owner"]["avatar_url"].str}""><span class="author"><a href="${pkg["owner"]["html_url"].str}">${pkg["owner"]["login"].str}</a></span>
+          </p>
+        <p>${pkg["update_age"].str}</p>
+        <button class="visit-pack-btn"><a href="/pkg/${pkg["name"].str}/"><i class="fas fa-link"> </i> Visit</a></button>
```
Comment (FedericoCeratto) [inline templates/home.tmpl]: A "visit" button is uncommon. The package name should be clickable - perhaps with a hint about it.

File: public/css/style.css [hunk 6]
```diff
@@ -1,117 +1,364 @@
-.section-header {
```
Comment (FedericoCeratto) [inline public/css/style.css]: The changes break the package page e.g. /pkg/jester - I suppose it's still work in progress.

File: templates/home.tmpl (hunk 2 above)
Comment (Willyboar) [reply templates/home.tmpl]: What did you mean false? The search box or you want to remove the install button?

File: public/css/style.css (hunk 6 above)
Comment (Willyboar) [reply public/css/style.css]: Yeap I 'll have to remove all the old css to test it properly.

File: templates/base.tmpl (hunk 1 above)
Comment (Willyboar) [reply templates/base.tmpl]: Yes, I saw it. Has a typo come from a paste.

File: templates/home.tmpl (hunk 5 above)
Comment (Willyboar) [reply templates/home.tmpl]: I put the link in the title. Do you want to remove the button totally?

File: templates/home.tmpl (hunk 2 above)
Comment (FedericoCeratto) [reply templates/home.tmpl]: "You don't have Nim in your computer. "

File: public/css/style.css (hunk 6 above)
Comment (FedericoCeratto) [reply public/css/style.css]: Please let me know when it's ready :)

File: templates/home.tmpl (hunk 2 above)
Comment (Willyboar) [reply templates/home.tmpl]: Ok, I have already removed that so this is resolved.

File: templates/home.tmpl (hunk 4 above)
Comment (Willyboar) [reply templates/home.tmpl]: I remove the install nim section, so I believe it is better now.

**Comment by dom96** [comment]
Screenshot?

**Comment by Willyboar** [comment]
> Screenshot?

nimble-directory

@dom96 This is the home page. But I don't like the lib showcase a lot so I have to change it.

**Comment by dom96** [comment]
wow, that looks amazing. Nice job!

**Comment by FedericoCeratto** [comment]
Merged, thanks!

**Pull request merged**